YC service discovery

ITD27M01 opened the pull request:
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1386

valyala commented on lib/promscrape/discovery/yandexcloud/api.go:
```diff
@@ -0,0 +1,272 @@
+
+func newAPIConfig(sdc *SDConfig, baseDir string) (*apiConfig, error) {
+	cfg := &apiConfig{
+		client: &http.Client{
```
Probably it would be better using the [discoveryutils.NewClient](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/5d364545bda670c03faab6de14400174d633cc6a/lib/promscrape/discoveryutils/client.go#L62) here if possible.

ITD27M01 replied:
Hi @valyala, thank you for the feedback. I need some time to investigate the possibility. m.b. you know is there any way to change the config of such client on the fly? yc api requires time limited bearer token and there `getFreshAPICredentials()` for the update

ITD27M01 replied:
I've found a way in Azure SD, I'll implement if it won't complicate the code.

ITD27M01 replied:
@valyala Hi, YC API uses per-service [edpoints](https://cloud.yandex.com/en-ru/docs/api-design-guide/concepts/endpoints), so it is difficult to use one [discoveryutils.NewClient](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/5d364545bda670c03faab6de14400174d633cc6a/lib/promscrape/discoveryutils/client.go#L62) for scraping, isn't it ?

valyala replied:
Agreed. Let's leave standard `http.Client` then here.

valyala reviewed (commented):
Looks great overall!

valyala commented:
@ITD27M01 , thanks for the pull request!

valyala commented:
@ITD27M01 , could you verify that the follow-up commit 83a4abda3f42b784efeddd6e71766a1a8b3b9cd2 doesn't break service discovery in Yandex Cloud?

ITD27M01 commented:
@valyala Works fine!

```
2022-08-04T20:45:23.039Z	info	VictoriaMetrics/lib/promscrape/discovery/yandexcloud/api.go:102	yandexcloud_sd: successfully refreshed service account api token; expiration: 41761.961 seconds
2022-08-04T20:45:24.176Z	info	VictoriaMetrics/lib/promscrape/discovery/yandexcloud/instance.go:33	yandexcloud_sd: collected 16 instances
2022-08-04T20:45:24.180Z	info	VictoriaMetrics/lib/promscrape/scraper.go:402	yandexcloud_sd_configs: added targets: 16, removed targets: 0; total targets: 16
```
